design/app-to-app.md

```diff
@@ -0,0 +1,87 @@
+App to App Interactions
+===========
+
+This document describes an extension to the SDK to allow MRE apps to receive content (e.g., photos) from the MRE clients. Content may be generated by the client itself, or, in future iterations not yet described here, may arrive from other apps with coordination from a particular client.
+
+## API Design
+This design introduces the concept of a `SharingMessage`, a base class of content-specific messages, such as `PhotoSharingMessage`. We provide an outline for the implementation of `PhotoSharingMessage` only, but expect to add more sharing capabilities over time.
+
+`SharingMessage` is itself relatively straightforward, but contains enough information to allow the recipient (callback listener) to cast it to the appropriate type (e.g., `PhotoSharingMessage`). We envision multiple message types and don't want to add a registration api for each.
+
+```ts
+class SharingMessage {
+    messageType: Readonly<string>;
+    sessionId: Readonly<string>;
+    userId: Readonly<Guid>;
+}
+
+class PhotoSharingMessage extends SharingMessage {
+    photoType: Readonly<string>; // Supported values: "jpg" and "png"
+    data: Readonly<Buffer>; // Raw bytes of an image file in the photoType format above
```

Comment:
If the photo can be large, I wonder if we would want to provide a thumbnail here with an async "get me the full file" function.  Maybe even a "get me the file at resolution X in format Y" function?  Goal being to enable the MRE to rapidly respond to photo sharing with some UI and to show progress indicators while the full transfer happens.

Reply:
I'll add plumbing for the size of the image, and allow the app to act on that in the `IsMessageDesired` stage. For now, skipping any app-direct resizing of the image.